nameable.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
+  end
```

Comment:
- [ ] Kindly change this method. You should trim the word if characters max than 10 characters. [Requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#:~:text=Implement%20a%20method%20correct_name%20that%20makes%20sure%20that%20the%20output%20of%20%40nameable.correct_name%20has%20a%20maximum%20of%2010%20characters.%20If%20it%27s%20longer%20it%20should%20trim%20the%20word.)
 _(if length>10 trim it otherwise keep the word.)_